File: solutions/javascript/move-zeros-to-the-left.js
let moveZerosToLeft = function(A) {
  if (A.length < 1) {
    return;
  }

  let lengthA = A.length;
  let writeIndex = lengthA - 1;
  let readIndex = lengthA - 1;

  while (readIndex >= 0) {
    if (A[readIndex] != 0) {
      A[writeIndex] = A[readIndex];
      writeIndex--;
    }
    
    readIndex--;
  }
  
  while (writeIndex >= 0) {
    A[writeIndex] = 0;
    writeIndex--;
  }
};

let v = [1, 10, 20, 0, 59, 63, 0, 88, 0];
console.log("Original Array: [" + v + "]");

moveZerosToLeft(v);

console.log("After Moving Zeros: [" + v+ "]");

File: solutions/python/move-zeros-to-the-left.py
def move_zeros_to_left(A):
  if len(A) < 1:
    return

  lengthA = len(A)
  write_index = lengthA - 1
  read_index = lengthA - 1

  while(read_index >= 0):
    if A[read_index] != 0:
      A[write_index] = A[read_index]
      write_index -= 1

    read_index -= 1

  while(write_index >= 0):
    A[write_index]=0;
    write_index-=1
    
v = [1, 10, 20, 0, 59, 63, 0, 88, 0]
print("Original Array:", v)

move_zeros_to_left(v)

print("After Moving Zeroes to Left: ", v)
